bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java

```diff
@@ -95,8 +109,12 @@ public ReadRowsRequest get() {
         }
 
         ReadHooks readHooks = new DefaultReadHooks();
-        ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
-        cachedRequest = readHooks.applyPreSendHook(builder.build());
+        Query query = Query.create(PLACEHOLDER_TABLE_ID);
+        RequestContext requestContext = RequestContext
+            .create(PLACEHOLDER_PROJECT_ID, PLACEHOLDER_INSTANCE_ID, PLACEHOLDER_APP_PROFILE_ID);
+        readHooks.applyPreSendHook(query);
+
+        cachedRequest = query.toProto(requestContext);
```

Comment:
Can we use placeholders for all four the values?

Then override it in [RequestWithTableNameValueProvider#get](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/2011/files#diff-5836fc24866e6a80ee2b663eb55592d4R347) with actual tableName & appProfileId.